ex9/js/main.js

```diff
@@ -0,0 +1,95 @@
+require(['ajaxGet'], function(ajaxGet) {
+  $(function() {
+    var on;
+    setEvents();
+    on = {
+      loadList: function() {
+         ajaxGet('json/data.json', function(phoneBookData) {
+            $('li').remove();
+            $("#phoneBookTemplate").tmpl( phoneBookData )
+              .appendTo( "#phoneBook" );
+          });
+      },
+      searchInput: function() {
+        console.log('searchInput');
+        if($('#textSearch').val() === '') {
+          $( "#btnFilter" ).prop( "disabled", true );
+        } else {
+          $( "#btnFilter" ).prop( "disabled", false );
+        }
+      },
+      searchChange: function() {
+        console.log('searchChange');
+        if($('#textSearch').val() === '') {
+          $( "#btnFilter" ).prop( "disabled", true );
+        } else {
+          $( "#btnFilter" ).prop( "disabled", false );
+        }
+      },
+      filter: function() {
+        console.log('filter click');
+      },
+      searchOn: function() {
+        console.log('searchOn');
+        // Search off
+        $( "#btnFilter" ).prop( "disabled", true );
```

Comment:
I would call searchChange instead of setting `disabled` directly.
Also, since you are not clearing the input after you hide and re-show it, it probably starts with the last search then, but the button is disabled.